### PR TITLE
Pdf extract priority

### PR DESCRIPTION
**Pull request by bjsi**
Allows you to set the priority of a PDF text / image extract.

1. Highlight text / images to extract:

image

2. Press Ctrl-Shift-Alt-X (Alt-Shift-X wouldn't work for me) and set the priority of the extract.

image

### CODE REVIEW

File: PDF/Viewer/IPDFViewer.SuperMemo.cs [hunk 1]
```diff
@@ -206,6 +212,25 @@ protected bool CreateSMExtract()
       return ret;
     }
 
+    protected async void CreateSMExtractWithPriority()
+    {
+
+      var result = await Show.Window()
+                         .For(new Prompt<double> { Title = "Extract Priority?", Value = -1 });
```
Comment (alexis-) [inline PDF/Viewer/IPDFViewer.SuperMemo.cs]: Could you:
- Set the default value to `Config.SMExtractPriority`
- Add this comment `// TODO: Create a better generic Prompt with: a) a description parameter, b) constraints parameters`

File: PDF/Viewer/IPDFViewer.SuperMemo.cs [hunk 2]
```diff
@@ -206,6 +212,25 @@ protected bool CreateSMExtract()
...
+      if (!result.Model.Confirmed)
+      {
+          return;
+      }
+
+      if (result.Model.Value < 0 || result.Model.Value > 100)
+      {
+          return;
```
Comment (alexis-) [inline PDF/Viewer/IPDFViewer.SuperMemo.cs]: There should be a warning for the user here

**Comment by bjsi** [comment]
Okay sorted. Here's an example:

image

Ctrl-Shift-Alt-X to open priority window. Opens with the default config priority.

image

User enters invalid priority.

image

**Comment by alexis-** [comment]
Looks like there's also a few conflicts here

**Comment by alexis-** [comment]
See https://github.com/supermemo/SuperMemoAssistant.Plugins.PDF/pull/114

**Pull request closed without merging**